eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/JsonNodeToClusteredVariantProcessor.java

```diff
@@ -0,0 +1,121 @@
+    private Function<IClusteredVariant, String> hashingFunction =
+            new ClusteredVariantSummaryFunction().andThen(new SHA1HashingFunction());
+
+    public JsonNodeToClusteredVariantProcessor() {
+    }
+
+    public DbsnpClusteredVariantEntity process(JsonNode jsonNode) {
+        return parseJsonNodeToDbsnpClusteredVariantEntity(jsonNode);
+    }
+
+    /**
+     * Parses the JSON into a DbsnpClusteredVariantEntity
+     * @param jsonRootNode root node of a dbSNP JSON line item
+     * @return dbsnpClusteredVariantEntity
+     */
+    private DbsnpClusteredVariantEntity parseJsonNodeToDbsnpClusteredVariantEntity(JsonNode jsonRootNode) {
+        long accession = jsonRootNode.path("refsnp_id").asLong();
+        ClusteredVariant clusteredVariant = parseJsonNodeToClusteredVariant(jsonRootNode);
+        String hashedMessage = hashingFunction.apply(clusteredVariant);
+        return new DbsnpClusteredVariantEntity(accession, hashedMessage, clusteredVariant);
+    }
+
+    private ClusteredVariant parseJsonNodeToClusteredVariant(JsonNode jsonRootNode) {
+        String assemblyAccession, contig;
+        assemblyAccession = contig = DEFAULT_VALUE_STRING;
+        // @see https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=9606
+        int taxonomyAccession = 9606;
+        long start = 0L;
+
+        // JSON date in ISO-8601 format
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-d'T'HH:mm'Z'");
+        LocalDateTime createdDate = LocalDateTime.parse(jsonRootNode.path("create_date").asText(), formatter);
+        VariantType type = prepareVariantType(jsonRootNode.path("primary_snapshot_data").path("variant_type").asText());
+        JsonNode infoNode = jsonRootNode.path("primary_snapshot_data").path("placements_with_allele");
+        for(JsonNode alleleInfo : infoNode) {
+            boolean isPtlp = alleleInfo.path("is_ptlp").asBoolean();
+            JsonNode assemblyInfo = alleleInfo.path("placement_annot").path("seq_id_traits_by_assembly");
+            if(assemblyInfo.size() == 0 || !isPtlp) {
+                continue;
+            }
+            assemblyAccession = assemblyInfo.get(0).path("assembly_accession").asText();
+            String alternateAllele, referenceAllele;
+            for (JsonNode alleleAndHgvs : alleleInfo.path("alleles")) {
+                JsonNode spdi = alleleAndHgvs.path("allele").path("spdi");
+                alternateAllele = spdi.path("inserted_sequence").asText();
+                referenceAllele = spdi.path("deleted_sequence").asText();
+                if(alternateAllele.equals(referenceAllele)) {
+                    contig = spdi.path("seq_id").asText();
+                    start = spdi.path("position").asLong(); //dbSNP stores in 0 base, EVA in 1 base
+                    return new ClusteredVariant(assemblyAccession, taxonomyAccession, contig, start,
```

Comment:
Since you now know how to filter for varianttype attribute, could you prepare a sample JSON with 5 values per variant type (filtered from one of the dbSNP 2 json files using [jq](https://stedolan.github.io/jq/)), run them through this pipeline and send the results?